dot/telemetry/telemetry.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2021 ChainSafe Systems (ON) Corp.
+// This file is part of gossamer.
+//
+// The gossamer library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The gossamer library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the gossamer library. If not, see <http://www.gnu.org/licenses/>.
+package telemetry
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"sync"
+	"time"
+
+	"github.com/gorilla/websocket"
+	log "github.com/sirupsen/logrus"
+)
+
+// Handler struct for holding telemetry related things
+type Handler struct {
+	buf             bytes.Buffer
+	wsConn          []*websocket.Conn
+	telemetryLogger *log.Entry
+}
+
+// MyJSONFormatter struct for defining JSON Formatter
+type MyJSONFormatter struct {
+}
+
+// Format function for handling JSON formatting, this overrides default logging formatter to remove
+//  log level, line number and timestamp
+func (f *MyJSONFormatter) Format(entry *log.Entry) ([]byte, error) {
+	serialized, err := json.Marshal(entry.Data)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal fields to JSON, %w", err)
+	}
+	return append(serialized, '\n'), nil
+}
+
+var once sync.Once
+var handlerInstance *Handler
+
+// GetInstance singleton pattern to for accessing TelemetryHandler
+func GetInstance() *Handler {
+	if handlerInstance == nil {
+		once.Do(
+			func() {
+				handlerInstance = &Handler{
+					buf: bytes.Buffer{},
+				}
+				log.SetOutput(&handlerInstance.buf)
+				log.SetFormatter(new(MyJSONFormatter))
+			})
+	}
+	return handlerInstance
+}
+
+// AddConnections adds connections to telemetry sever
+func (h *Handler) AddConnections(conns []interface{}) {
```

Comment:
Why is the param type interface? Interface is handy when the params could be of any type.
But here the params type is fixed.

Reply:
I updated this to use a struct as parameter.  I was using []interface{} because this is the result from unmarshalling the genesis.json, but yes it makes more sense to use this value as a struct instead.